test/basic_test.go

```diff
@@ -362,6 +362,40 @@ func TestPubSubWithReply(t *testing.T) {
 	}
 }
 
+func TestMsgRespond(t *testing.T) {
+	s := RunDefaultServer()
+	defer s.Shutdown()
+	nc := NewDefaultConnection(t)
+	defer nc.Close()
+
+	m := &nats.Msg{}
+	if err := m.Respond(nil); err != nats.ErrMsgNotBound {
+		t.Fatal("Expected ErrMsgNotBound error")
+	}
+
+	sub, err := nc.Subscribe("req", func(msg *nats.Msg) {
+		msg.Respond([]byte("42"))
+	})
+	if err != nil {
+		t.Fatal("Failed to subscribe: ", err)
+	}
+
+	// Fake the bound notion by assigning Sub directly to test no reply.
```

Comment:
I would have simply created a sync sub on another subject, publish a message (not request) to that subject, get the message, then call respond. No need to fake it.

Reply:
Its public so easy to do, but if you feel strongly I could make legit here.